tests/unit/storage/data/DummyService.js

```diff
@@ -133,6 +135,38 @@ class DummyService {
         objStream.pipe(writeStream);
         return callback();
     }
+    putObjectTagging(tagParams, callback) {
+        if (tagParams.Key === 'externalBackendTestBucket/externalBackendMissingKey') {
+            const err = errors.NoSuchKey;
+            return callback(err);
+        }
+
+        const keys = Object.keys(tagParams);
+        assert(keys.length > 0);
+        assert(tagParams.Tagging.TagSet.length > 0);
+        tagParams.Tagging.TagSet.forEach(tag => {
+            assert(tag.Key.length > 0);
+            assert(tag.Value.length > 0);
+        });
+
+        if (tagParams.VersionId) {
+            assert.strictEqual(tagParams.VersionId, 'latestversion');
```

Comment:
Why do you make this assertion here?

Reply:
to make sure we send the versionId, in case one day we remove it (like I was about to do before you came to me)